R/uvhydrograph-render.R

```diff
@@ -363,3 +352,180 @@ YEndpoint <- function (corr.value.sequence, uncorr.value.sequence) {
+          ),
+      appr_working_dv = list(
+          points = append(list(x=x, y=y, pch=reportObject[[1]]$point_type, legend.name=legend.name), styles$appr_working_dv_points)
+          )
+  )
+  
+  return(plotConfig)
+}
+
+#' TODO
+#' @importFrom grDevices rgb
+getSecondaryPlotConfig <- function(reportObject, info, correctionLabels) {
+  styles <- getUvStyles()
+  
+  x <- reportObject[[1]]$time
+  y <- reportObject[[1]]$value
+  
+  plotStartDate <- info$plotStartDate
+  plotEndDate <- info$plotEndDate
+  
+  corrArrowPositions <- getCorrectionArrowPositions(x, y, correctionLabels)
+  corrAblinePositions <- getCorrectionAbLines(reportObject)
+  
+  secondary_lbl <- info$secondary_lbl
+  plotConfig <- switch(names(reportObject),
+      corr_UV2 = list(
+          lines = append(list(x=x,y=y, legend.name=paste(styles$corr_UV2_lbl, secondary_lbl)), styles$corr_UV2_lines)
+          ), 
+      est_UV2 = list(
+          lines = append(list(x=x,y=y,legend.name=paste(styles$est_UV2_lbl, secondary_lbl)), styles$est_UV2_lines)
+          ),
+      uncorr_UV2 = list(
+          lines = append(list(x=x,y=y, legend.name=paste(styles$uncorr_UV2_lbl, secondary_lbl)), styles$uncorr_UV2_lines)
+          ),                
+      effect_shift = list(
+          lines=append(list(x=x,y=y, legend.name=paste(secondary_lbl, info$tertiary_lbl)), styles$effect_shift_lines),
+          text=append(list(x=x[1], y=y[1]), styles$effect_shift_text)
+          ),
+      series_corr2 = list(
+          lines=append(list(x=0, y=0, xlim = c(plotStartDate, plotEndDate)), styles$series_corr2_lines),
+          abline=append(list(v=corrAblinePositions$time, legend.name=paste(styles$series_corr2_correction_lbl, secondary_lbl)), styles$series_corr2_ablines),
+          arrows=append(list(x0=corrArrowPositions$xorigin, x1=corrArrowPositions$x, y0=corrArrowPositions$y, y1=corrArrowPositions$y), styles$series_corr2_arrows),
+          points=append(list(x=correctionLabels$x, y=correctionLabels$y, cex=correctionLabels$r), styles$series_corr2_points),
+          text=append(list(x=correctionLabels$x, y=correctionLabels$y, labels=correctionLabels$label), styles$series_corr2_text)
+          ),
+      gage_height = list(
+          points=append(list(x=x, y=y), styles$gage_height_points),
+          callouts=list(x=x, y=y, labels=reportObject$gage_height$n)
+          ),
+      gw_level = list(points = append(list(x=x,y=y), styles$gw_level_points)), 
+      meas_shift = list(
+          points=append(list(x=x, y=y), styles$meas_shift_points),
+          error_bar=append(list(x=x, y=y, y.low=(y-reportObject$meas_shift$minShift), y.high=(reportObject$meas_shift$maxShift-y)), styles$meas_shift_error_bars)
+          )
+  )
+  
+  return(plotConfig)
+}
+
+#' TODO
+getCorrectionArrowPositions <- function(time, value, correctionLabels) {
+  x <- time
+  y <- value
+  
+  corrArrowPositions <- list()
```

Comment:
It's actually considered good practice to always initialize variables to their desired default variables. Reasoning being if the logic which comes after does not hit all edge cases you still get the default (eg: if there are nested if/else statements where someone leaves out an edge case, we still get the default value).